### PR TITLE
add homework 3

### PR DESCRIPTION
**Pull request by NataliKokovina**

### CODE REVIEW

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-shadow */
+
+// 1.1 Exponentiation
+
+function Exponentiation(testNumber) {
+    console.log(testNumber ** 2);
```
Comment (markoffden) [inline homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: the condition is not to use `**` exponential operators

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+array(50, 5);
+array(32, 4);
+
+//  1.3 Exponentiation elements of arr
+
+function Expon(numSet) {
+    const arrExp = [];
+    for (let i = 0; i < numSet.length; i++) {
+        arrExp.push(numSet[i] ** 2);
```
Comment (markoffden) [inline homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: again, no`**` operator please

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    console.log(arr);
+    return (arr);
+}
+array(50, 5);
+array(32, 4);
+
+//  1.3 Exponentiation elements of arr
+
+function Expon(numSet) {
```
Comment (markoffden) [inline homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: also, no capitalized function naming unless it's a constructor

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        arrExp.push(numSet[i] ** 2);
+    }
+    console.log(arrExp);
+    return (arrExp);
+}
+Expon(array(32, 4));
+
+// 1.4 Sum elements
+
+function Sum(arrForSumm) {
```
Comment (markoffden) [inline homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: naming

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/recursion.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+function recursion(fA, sA) {
+    if (fA <= sA) {
+        for (let i = 0; i < sA; i++) {
+            if (2 ** (i + 1) === fA) {
```
Comment (markoffden) [inline homeworks/natalia.kokovina_NataliKokovina/homework_3/recursion.js]: no `**` here as well

Comment (NataliKokovina) [reply homeworks/natalia.kokovina_NataliKokovina/homework_3/recursion.js]: fix

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js (hunk 4 above)
Comment (NataliKokovina) [reply homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: fix

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js (hunk 3 above)
Comment (NataliKokovina) [reply homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: fix

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js (hunk 2 above)
Comment (NataliKokovina) [reply homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: fix

File: homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js (hunk 1 above)
Comment (NataliKokovina) [reply homeworks/natalia.kokovina_NataliKokovina/homework_3/function.js]: fix

**Comment by markoffden** [comment]
accepted, but may be optimized:

1. in tasks 1.3 and 1.4 you could utilize `map()` and `reduce()` methods accordingly

2. your recursion function could simply return true/false instead of 1/0, it's more readable

pay attention to such things in future

**Pull request merged**